File: 1556-make-two-arrays-equal-by-reversing-subarrays/1556-make-two-arrays-equal-by-reversing-subarrays.kt
class Solution {
    fun canBeEqual(target: IntArray, arr: IntArray): Boolean {
        val mp : Array<Int> = Array(1000) { 0 };
        for (n in target) mp[n - 1]++;
        for (n in arr) mp[n - 1]--;
        for (k in mp) if (k != 0) return false;
        return true;
    }
}

File: 1556-make-two-arrays-equal-by-reversing-subarrays/1556-make-two-arrays-equal-by-reversing-subarrays.c
bool canBeEqual(int* target, int targetSize, int* arr, int arrSize) {
    int mp[1001] = {0};
    while (targetSize--) mp[*target++]++;
    while (arrSize--) mp[*arr++]--;
    for (int i = 1; i < 1001; i++) if (mp[i]) return false;
    return true;
}